wallet/src/de/schildbach/wallet/ui/transactions/TxResourceMapper.java

```diff
@@ -1,23 +1,50 @@
-package de.schildbach.wallet.util;
+/*
+ * Copyright 2019 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.wallet.ui.transactions;
+
+import android.text.format.DateUtils;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.StringRes;
 
 import org.bitcoinj.core.Coin;
+import org.bitcoinj.core.Context;
 import org.bitcoinj.core.RejectedTransactionException;
 import org.bitcoinj.core.Transaction;
+import org.bitcoinj.core.TransactionBag;
 import org.bitcoinj.core.TransactionConfidence;
 import org.bitcoinj.wallet.Wallet;
+import org.dash.wallet.common.transactions.TransactionUtils;
 
 import de.schildbach.wallet.Constants;
+import de.schildbach.wallet.util.WalletUtils;
 import de.schildbach.wallet_test.R;
 
-public class TransactionUtil {
+public class TxResourceMapper {
```

Comment:
This class is largely unchanged but renamed to reflect its purpose.
Other classes can be derived from it (like "CrowdNodeTxResourceMapper") to provide custom titles/statuses